src/Examples/Advanced/CustomProperties/custom_physical_property_packages.ipynb

```diff
@@ -0,0 +1,1329 @@
+    "    \"\"\"\n",
+    "\n",
+    "    def build(self):\n",
+    "        \"\"\"Callable method for Block construction.\"\"\"\n",
+    "        super(HDAStateBlockData, self).build()\n",
+    "\n",
+    "        add_state_variables(self)\n",
+    "        add_mole_fraction_constraint(self)\n",
+    "        add_molecular_weight_and_density(self)\n",
+    "        \n",
+    "    def _enth_mol(self):\n",
+    "        add_enth_mol(self)\n",
+    "\n",
+    "    def define_state_vars(self):\n",
+    "        return return_state_var_dict(self)\n",
+    "        \n",
+    "    def get_material_flow_terms(self, p, j):\n",
+    "        return self.flow_mol * self.mole_frac_comp[j]\n",
+    "\n",
+    "    def get_enthalpy_flow_terms(self, p):\n",
+    "        \"\"\"Create enthalpy flow terms.\"\"\"\n",
+    "        return self.flow_mol * self.enth_mol\n",
+    "\n",
+    "    def default_material_balance_type(self):\n",
+    "        return MaterialBalanceType.componentPhase\n",
+    "\n",
+    "    def default_energy_balance_type(self):\n",
+    "        return EnergyBalanceType.enthalpyTotal\n",
+    "\n",
+    "    def get_material_flow_basis(self):\n",
+    "        return MaterialFlowBasis.molar"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Demonstration\n",
+    "\n",
+    "We have now finished writing our example property package, so let us put it to use in a simple demonstration.\n",
+    "\n",
+    "First, we need to import some basic components from `pyomo.environ` and `idaes.core` to build a flowsheet model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyomo.environ import ConcreteModel, SolverFactory\n",
+    "from idaes.core import FlowsheetBlock"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we create `ConcreteModel` and a steady-state `FlowsheetBlock` to contain our example, ot which we attach an instance of our new `HDAPhysicalParameterBlock`.\n",
+    "\n",
+    "We can then create an instance of the `HDAStateBlock` directly from the parameter block using the `build_state_block` method. This uses the reference to the State Block class that we attached ot the Physical Parameter Block earlier in the example to automatically create an instance of the correct class. Note that we index the State Block by the time domain, so that it looks like a typical state block in a flowsheet, and we set `defined_state = True` so that we can set values for all the component mole fractions later."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "\n",
+    "m.fs = FlowsheetBlock(default={\"dynamic\": False})\n",
+    "\n",
+    "m.fs.thermo_props = HDAParameterBlock()\n",
+    "\n",
+    "m.fs.state = m.fs.thermo_props.build_state_block(\n",
```

Comment:
Here, in code block 17, I hit an error: 
`PropertyNotSupportedError: fs.state[0.0] package property metadata method does not contain a method for component_list. Please select a package which supports the necessary properties for your process.`

Reply:
The traceback is more extensive but didn't want to leave such a verbose comment.

Reply:
I don't get an issue when I run the example. Could you check your IDAES version first of all, and also send me the full traceback (Slack is fine).

Reply:
Resolved this issue by using a later version of IDAES, but now the last cell gives an error; it appears that the RPP property package no longer exists (instead I see RPP3, RPP4, and RPP5 in  `idaes/generic_models/properties/core/pure` 

I'd say that incorporating `doctest` would be useful, except I'm not sure how that would work for a jupyter notebook; I do know how to work doctest with .rst files though.

Reply:
We have other ways of testing Jupyter notebooks, and we will include some test cells in the final version.

Regarding the `RPP` module, I probably updated that on my local branch and it is a small change.